client/sites-dashboard-v2/sites-dashboard.tsx

```diff
@@ -133,6 +133,11 @@ const SitesDashboardV2 = ( {
 		queryParamKeys: [ 'page', 'per-page', 'status', 'search' ],
 	} );
 
+	// If calypso state changes the selected site, ensure the dataViewsState is updated as well.
+	useEffect( () => {
```

Comment:
Nice finding!

Could we move this logic to [useSyncSelectedSite()](https://github.com/Automattic/wp-calypso/blob/32b2cc2404354a182dbcc514da2d62573ee234b8/client/sites-dashboard-v2/hooks/use-sync-selected-site.ts)? Would be nice to colocate related/similar logic together.

Reply:
makes sense to me!